Theme.h

```diff
@@ -0,0 +1,488 @@
+			{ImGuiCol_TabUnfocusedActive, ImVec4(0.3602f, 0.1633f, 0.0478f, 1.0000f)},
+			{ImGuiCol_DockingPreview, ImVec4(0.9800f, 0.5876f, 0.2600f, 0.7000f)},
+			{ImGuiCol_DockingEmptyBg, ImVec4(0.2000f, 0.2000f, 0.2000f, 1.0000f)},
+			{ImGuiCol_PlotLines, ImVec4(0.6100f, 0.6100f, 0.6100f, 1.0000f)},
+			{ImGuiCol_PlotLinesHovered, ImVec4(1.0000f, 0.4300f, 0.3500f, 1.0000f)},
+			{ImGuiCol_PlotHistogram, ImVec4(0.9000f, 0.7000f, 0.0000f, 1.0000f)},
+			{ImGuiCol_PlotHistogramHovered, ImVec4(1.0000f, 0.6000f, 0.0000f, 1.0000f)},
+			{ImGuiCol_TableHeaderBg, ImVec4(0.0759f, 0.0748f, 0.0737f, 1.0000f)},
+			{ImGuiCol_TableBorderStrong, ImVec4(1.0000f, 0.0000f, 0.0000f, 1.0000f)},
+			{ImGuiCol_TableBorderLight, ImVec4(1.0000f, 0.0000f, 0.0000f, 1.0000f)},
+			{ImGuiCol_TableRowBg, ImVec4(0.0000f, 0.0000f, 0.0000f, 0.0000f)},
+			{ImGuiCol_TableRowBgAlt, ImVec4(1.0000f, 1.0000f, 1.0000f, 0.0600f)},
+			{ImGuiCol_TextSelectedBg, ImVec4(0.9800f, 0.7309f, 0.2600f, 0.3500f)},
+			{ImGuiCol_DragDropTarget, ImVec4(1.0000f, 1.0000f, 0.0000f, 0.9000f)},
+			{ImGuiCol_NavHighlight, ImVec4(0.9145f, 0.0921f, 0.2133f, 1.0000f)},
+			{ImGuiCol_NavWindowingHighlight, ImVec4(1.0000f, 1.0000f, 1.0000f, 0.7000f)},
+			{ImGuiCol_NavWindowingDimBg, ImVec4(0.8000f, 0.8000f, 0.8000f, 0.2000f)},
+			{ImGuiCol_ModalWindowDimBg, ImVec4(0.8000f, 0.8000f, 0.8000f, 0.3500f)},
+		},
+		{ 
+		}
+	}},
+};
+
+#pragma endregion
+
+#pragma region Theme Functions
+
+/**
+* @brief Applies the Selected ThemeSet to the ImGui Window using ImGui::PushStyleColor and ImGui::PushStyleVar
+* 
+* @param const std::string& themeName The name of the ThemeSet to Load
+* @return int The number of colors pushed + (the number of styles pushed << 16)
+*/
+int PushTheme(const std::string& themeName)
+{
+	int colorCount = 0;
+	int styleCount = 0;
+
+	auto themeIter = themeSet.find(themeName);
+	if (themeIter != themeSet.end()) {
+		const Theme& theme = themeIter->second;
+		for (const auto& color : theme.colors)
+		{
+			ImGui::PushStyleColor(color.property, color.color);
+			++colorCount;
+		}
+		for (const auto& style : theme.styles)
+		{
+			if (std::holds_alternative<ImVec2>(style.value)) {
+				ImGui::PushStyleVar(style.property, std::get<ImVec2>(style.value));
+			}
+			else if (std::holds_alternative<float>(style.value)) {
+				ImGui::PushStyleVar(style.property, std::get<float>(style.value));
+			}
+			else if (std::holds_alternative<int>(style.value)) {
+				ImGui::PushStyleVar(style.property, static_cast<float>(std::get<int>(style.value)));
+			}
+			++styleCount;
+		}
+	}
+
+	return colorCount + (styleCount << 16);
+}
+
+/**
+* @brief Pop a value from the ImGui context
+* @param int popCounts The number of colors pushed + (the number of styles pushed << 16)
+*/
+void PopTheme(int popCounts)
+{
+	int colorCount = popCounts & 0xFFFF;
+	int styleCount = popCounts >> 16;
+	ImGui::PopStyleColor(colorCount);
+	ImGui::PopStyleVar(styleCount);
+}
```

Comment:
just pass two ints, or a struct with two uint16s, this usage is not intuitive and buys you nothing.